hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateJMSMessageHeader.java

```diff
@@ -38,10 +38,10 @@ public void populateJMSRequestHeader(Exchange exchange, String v2Message) throws
 		} catch (CSIException e) {
 			logger.error("{} - TransactionId: {}, Exception while converting hexadecimal message Control Id to byte {}",
 				methodName, exchange.getExchangeId(), e.getMessage());
-        	throw e;
+        	throw new IllegalArgumentException("Exception while converting hexadecimal message Control Id to byte");
 		}
 		
-		String exchangeId = exchange.getExchangeId().replaceAll("-", "");
+		String exchangeId = exchange.getExchangeId().replace("-", "");
```

Comment:
This changes the business logic. Is this required?

Reply:
https://rules.sonarsource.com/java/RSPEC-5361